drizzlepac/haputils/processing_utils.py

```diff
@@ -341,6 +341,53 @@ def compute_sregion(image, extname='SCI'):
     if closefits:
         hdu.close()
 
+def add_skycell_to_header(image_filename, extname='SCI'):
+    """Determines the skycells for which the image falls within and adds the 
+    information to the science header.
+
+    Parameters
+    ----------
+    image : str 
+        Image filename to update with the skycell keyword in each extname extension.
+    extname : str, optional
+        EXTNAME value for extension containing the header to be updated
+        
+    Returns
+    -------
+    Nothing; image header has been updated unless keyword already exists.
+    """    
+
+    # Open image and determine whether to consequently close it
+    try: 
+        hdu, closefits = _process_input(image_filename)
+    except:
+        log.error(f"Could not open {image_filename} during add_skycell_to_header. Exiting.")
+
+    # Find all extensions to be updated
+    numext = countExtn(hdu, extname=extname)
+    
+    for extnum in range(1, numext + 1):
+        sciext = (extname, extnum)
+        # only add skycell keyword if it does not already exist
```

Comment:
This is just FYI and you do not have to change as I see the variable in question, _extnum_, has the same variable name in other routines in this module.  Technically, this variable would be more appropriately named _extver_.  An _extnum_ is just an extension number which is sequential (1, 2, 3, ...) in the FITS file.  An _extver_ is associated with an imset and this is what you are really updating.   The way this is referenced is [extname, extver] or ['sci', 1] or ['err', 5] where ['err', 5] could be the 15th extension.

Reply:
I like the suggestion and I've made the change anyway.